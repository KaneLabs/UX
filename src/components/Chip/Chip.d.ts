export default Chip;
declare function Chip({ text, flat, style, hoverStyle, focusStyle, children, ...rest }: {
    [x: string]: any;
    text?: any;
    flat?: boolean | undefined;
    style?: any;
    hoverStyle?: any;
    focusStyle?: any;
    children: any;
}): JSX.Element;
