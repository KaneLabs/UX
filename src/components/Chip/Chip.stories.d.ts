declare const _default: import("@storybook/addons").StoryApi<import("@storybook/react/dist/client/preview/types").StoryFnReactReturnType>;
export default _default;
