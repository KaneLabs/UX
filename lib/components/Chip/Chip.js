var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _slicedToArray2=_interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));var _objectWithoutProperties2=_interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));var _react=_interopRequireWildcard(require("react"));var _reactNative=require("react-native");var _theme=require("../../theme");var _Paper=_interopRequireDefault(require("../Paper"));var _Typography=_interopRequireDefault(require("../Typography"));var _this=this,_jsxFileName="/Users/kane/Lab/Eros/src/components/Chip/Chip.tsx";var DEFAULT_CHIP_HEIGHT=32;var Chip=function Chip(props){var text=props.text,_props$flat=props.flat,flat=_props$flat===void 0?false:_props$flat,style=props.style,hoverStyle=props.hoverStyle,focusStyle=props.focusStyle,children=props.children,rest=(0,_objectWithoutProperties2.default)(props,["text","flat","style","hoverStyle","focusStyle","children"]);var _useState=(0,_react.useState)(false),_useState2=(0,_slicedToArray2.default)(_useState,2),focus=_useState2[0],setFocus=_useState2[1];var _useState3=(0,_react.useState)(false),_useState4=(0,_slicedToArray2.default)(_useState3,2),hover=_useState4[0],setHover=_useState4[1];var styles=useStyles();var ChipStyle=_react.default.useMemo(function(){return _reactNative.StyleSheet.flatten([styles.Chip,hover&&styles.ChipHovered,hover&&hoverStyle,focus&&styles.ChipFocused,focus&&focusStyle,flat&&styles.ChipFlat,_reactNative.Platform.OS==='web'&&{outlineWidth:0},style]);},[]);return _react.default.createElement(_Paper.default,(0,_extends2.default)({style:ChipStyle,onFocus:function onFocus(){return setFocus(true);},onBlur:function onBlur(){return setFocus(false);},onMouseEnter:function onMouseEnter(){return setHover(true);},onMouseLeave:function onMouseLeave(){return setHover(false);}},rest,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:50,columnNumber:5}}),text&&_react.default.createElement(_Typography.default,{type:_theme.TypographyTypes.body2,__self:_this,__source:{fileName:_jsxFileName,lineNumber:57,columnNumber:17}},text)||children||null);};var useStyles=(0,_theme.makeStyles)(function(theme){return{Chip:_reactNative.Platform.select({default:{height:DEFAULT_CHIP_HEIGHT,borderRadius:DEFAULT_CHIP_HEIGHT,borderColor:theme.borderColor,borderWidth:theme.borderWidth,paddingLeft:theme.padding,paddingRight:theme.padding,overflow:'hidden',justifyContent:'center'}}),ChipFlat:_reactNative.Platform.select({web:{backgroundColor:theme.canvas2Opaque,borderWidth:0,outlineWidth:0},default:{backgroundColor:theme.canvas2Opaque,borderWidth:0}}),ChipHovered:(0,_extends2.default)({borderColor:theme.textColor.disabled,color:theme.textColor.secondary},theme.shadow(8),{backgroundColor:theme.canvas2Opaque}),ChipFocused:(0,_extends2.default)({borderColor:theme.textColor.secondary,color:theme.textColor.primary},theme.shadow(8),{backgroundColor:theme.canvas2Opaque})};});var _default=Chip;exports.default=_default;