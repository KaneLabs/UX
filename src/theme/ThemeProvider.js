import React, {
  useState, useMemo, createContext, useCallback,
} from 'react';
import { useColorScheme } from 'react-native';
import * as light from './light';
import * as dark from './dark';

export const ThemeContext = createContext(null);

export const ThemeProvider = (props) => {
  const { children, initialScheme } = props;
  const colorScheme = useColorScheme();
  console.log(`Detected device color scheme: ${colorScheme}`);
  const [themeMode, setThemeMode] = useState(initialScheme || colorScheme);
  const toggleTheme = useCallback(() => setThemeMode((mode) => (mode === 'light' ? 'dark' : 'light')));
  const value = useMemo(
    () => (themeMode === 'light' ? [light, toggleTheme] : [dark, toggleTheme]),
    [themeMode],
  );

  return (
    <ThemeContext.Provider value={value}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;
