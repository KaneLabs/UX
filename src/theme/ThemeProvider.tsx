import * as React from 'react';
import { useColorScheme, useWindowDimensions } from 'react-native';
import * as light from './light';
import * as dark from './dark';
import Theme from './Theme';
import makeTheme, { ThemeModes } from './makeTheme';

const defaultTheme: Theme = makeTheme({ mode: ThemeModes.light });
const defaultContext: ThemeContextType = [defaultTheme, () => null];
export const ThemeContext = React.createContext(defaultContext);

export interface ThemeOverrideProps {
  primaryColor?: string;
  secondaryColor?: string;
}

export interface ThemeProviderProps {
  children: React.ReactNode;
  initialScheme?: ThemeModes;
  theme?: ThemeOverrideProps;
}

export type ThemeContextType = [Theme, () => any];

export type ThemeProvider<ThemeProviderProps> = (
  props: ThemeProviderProps,
) => React.ContextType<any>;

export const ThemeProvider = (props: ThemeProviderProps) => {
  const { children, initialScheme } = props;
  if (!props.theme) console.log('No Theme provided, using default');
  const colorScheme = useColorScheme();

  const [themeMode, setThemeMode] = React.useState(
    initialScheme || colorScheme || 'dark',
  );

  const window = useWindowDimensions();

  const toggleTheme = React.useCallback(
    () =>
      setThemeMode((mode) =>
        mode === ThemeModes.light ? ThemeModes.dark : ThemeModes.light,
      ),
    [],
  );

  const theme = React.useMemo(() => {
    const initialTheme = props.theme ?? {};
    const theme =
      themeMode === ThemeModes.light
        ? makeTheme({ mode: ThemeModes.light, ...initialTheme })
        : makeTheme({ mode: ThemeModes.dark, ...initialTheme });
        theme.window = window;
    return theme;
  }, [themeMode, props.theme, window, colorScheme]);

  const value = [theme, toggleTheme] as ThemeContextType;
  return (
    <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>
  );
};

export default ThemeProvider;
