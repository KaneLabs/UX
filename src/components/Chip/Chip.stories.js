import React from 'react';
import { View } from 'react-native';

import Chip from './Chip';

import Container from 'eros-ui-storybook/components/Container';
import { storiesOf } from 'eros-ui-storybook/helpers/storiesOf';

export default storiesOf('Components|Chip', module)
  .addParameters({ jest: ['Chip'] })
  .add('Chip', () => (
    <Container>
      <View style={{ flexDirection: 'column', flexWrap: 'wrap' }}>
        <Chip text="Default" />
      </View>
    </Container>
  ));
