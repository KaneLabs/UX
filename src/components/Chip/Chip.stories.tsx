import React from 'react';
import { View } from 'react-native';

import Chip from './Chip';

import Container from '@kanelabs/ux-storybook/components/Container';
import { storiesOf } from '@kanelabs/ux-storybook/helpers/storiesOf';

export default storiesOf('Components|Chip', module)
  .addParameters({ jest: ['Chip'] })
  .add('Chip', () => (
    <Container>
      <View style={{ flexDirection: 'column', flexWrap: 'wrap' }}>
        <Chip text="Default" />
      </View>
    </Container>
  ));
