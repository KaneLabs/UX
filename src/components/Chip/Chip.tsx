import React, { useState } from 'react';
import { View, Platform } from 'react-native';

import { useTheme, makeStyles, shadow } from 'eros-ui/theme';
import Paper from 'eros-ui/components/Paper';
import Typography from 'eros-ui/components/Typography';

const DEFAULT_CHIP_HEIGHT = 32;

const useStyles = makeStyles(theme => ({
  default: Platform.select({
    default: {
      height: DEFAULT_CHIP_HEIGHT,
      borderRadius: DEFAULT_CHIP_HEIGHT,
      // maxWidth: DEFAULT_CHIP_HEIGHT * 3,
      borderColor: theme.borderColor,
      borderWidth: theme.borderWidth,
      paddingLeft: theme.padding,
      paddingRight: theme.padding,
      overflow: 'hidden',
      justifyContent: 'center',
    },
  }),
  flat: Platform.select({
    web: {
      backgroundColor: theme.canvas2Opaque,
      borderWidth: 0,
      outlineWidth: 0,
    },
    default: { backgroundColor: theme.canvas2Opaque, borderWidth: 0 },
  }),
}));

const Chip = ({
  text = null,
  flat = false,
  style = null,
  hoverStyle = null,
  focusStyle = null,
  children,
  ...rest
}) => {
  const [{ textColor, canvas2Opaque }] = useTheme();
  const [focus, setFocus] = useState(false);
  const [hover, setHover] = useState(false);
  const styles = useStyles();

  const ChipStyle = [
    styles.default,
    hover && {
      borderColor: textColor.disabled,
      color: textColor.secondary,
      ...shadow(8),
      backgroundColor: canvas2Opaque,
      ...hoverStyle,
    },
    // focus && {
    //   borderColor: textColor.secondary,
    //   color: textColor.primary,
    //   ...shadow(8),
    //   backgroundColor: canvas2Opaque,
    //   ...focusStyle,
    // },
    flat && styles.flat,
    Platform.OS === 'web' && { outlineWidth: 0 }, // needed for Web
    style,
  ];

  return (
    <Paper
      style={ChipStyle}
      onFocus={() => setFocus(true)}
      onBlur={() => setFocus(false)}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
      {...rest}>
      {(text && <Typography type="body2">{text}</Typography>) ||
        children ||
        null}
    </Paper>
  );
};

export default Chip;
