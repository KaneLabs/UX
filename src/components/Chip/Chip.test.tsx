import React from 'react';
import renderer from 'react-test-renderer';
import Chip from './Chip';
import ThemeProvider from '../../theme/ThemeProvider';

test('Chip', () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <Chip text="Default" />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
