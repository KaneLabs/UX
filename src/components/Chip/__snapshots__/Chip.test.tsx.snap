// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip 1`] = `
<View
  accessible={true}
  focusable={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255,255,255, 1)",
      "borderColor": "rgba(0,0,0,0.12)",
      "borderRadius": 32,
      "borderWidth": 0.5,
      "height": 32,
      "justifyContent": "center",
      "overflow": "hidden",
      "paddingLeft": 12,
      "paddingRight": 12,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(0,0,0,0.87)",
        "fontSize": 14,
        "fontWeight": "300",
        "letterSpacing": 0.25,
        "lineHeight": 17,
        "marginBottom": 0,
      }
    }
  >
    Default
  </Text>
</View>
`;
